engine/engine/src/engine.cpp

```diff
@@ -1798,8 +1801,14 @@ namespace dmEngine
                         return;
                     }
                 }
-
-                dmJobThread::Update(engine->m_JobThreadContext);
+#if defined(DM_HAS_THREADS)
+                uint64_t jobthread_max_time_us = 0;
+#else
+                // Dictates max time the job thread may take.
+                // Note that it will always process at least one item if available
+                uint64_t jobthread_max_time_us = 3 * 1000;
```

Comment:
May well be configurable later. Also, should it be shared between the different job thread contexts?

Side note:
We currently create multiple job thread contexts, but each context may have 0, 1 or more threads.
So, in terms of keeping the threads contained, it probably makes sense to use 1 job thread context for all our internal systems.

Reply:
My opinion is every job context should have their own `max_time` setting. And also I think that it's good to have multiple job_contexts because different subsystems can have different type of jobs. I mean one subsystem can have lots of small jobs, other - small amount of long-running jobs. I think if we have multiple job context it helps to tweak each job context according to job types requirements.

Reply:
Yeah, I think overall, it generally needs more work if we are to mix different work types on the same thread.
E.g. having different priorities on jobs, max time per type etc.
Scheduling is a tricky thing.
So far, we've been reacting to the tasks, but I feel we soon need to model this system after some exsiting job system out in the wild. TBD.